src/solver/include/toolkitAPI.h

```diff
@@ -0,0 +1,561 @@
+
+
+*/
+#ifndef TOOLKITAPI_H
+#define TOOLKITAPI_H
+
+#ifdef WINDOWS
+#ifdef __MINGW32__
+#define DLLEXPORT __declspec(dllexport) __cdecl
+#else
+#define DLLEXPORT __declspec(dllexport) __stdcall
+#endif
+#else
+#define DLLEXPORT
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define _CRT_SECURE_NO_DEPRECATE
+
+#include "toolkitAPI_enums.h"
+
+
+/**
+ @brief Opens SWMM input file, reads in network data, runs, and closes
+ @param f1 pointer to name of input file (must exist)
+ @param f2 pointer to name of report file (to be created)
+ @param f3 pointer to name of binary output file (to be created)
+ @param pointer to callback function (for printing progress)
+ @return error code
+*/
+int  DLLEXPORT   swmm_run_cb(const char *f1, const char *f2, const char *f3,
+    void (*callback) (double *));
+
+/**
+ @brief Get the text of an error code.
+ @param errcode The error code
+ @param[out] errorMsg The error string represented by the code
+ @return Error code
+*/
+int DLLEXPORT swmm_getAPIError(int ErrorCodeAPI, char **errorMsg);
+
+/**
+ @brief Finds the index of an object given its ID.
+ @param type An object type
+ @param id The object ID
+
+ @param[out] index The objects index
+ @return Error code
+*/
+int DLLEXPORT swmm_project_findObject(int type, char *id, int *index);
+
+/**
+@brief Gets Simulation Unit
+@param type Option code (see @ref SM_Units)
+@param[out] value Option value
+@return Error code
+*/
+int DLLEXPORT swmm_getSimulationUnit(int type, int *value);
+
+/**
+ @brief Gets Simulation Analysis Setting
+ @param type Option code (see @ref SM_SimOption)
+ @param[out] value Option value
+ @return Error code
+ */
+int DLLEXPORT swmm_getSimulationAnalysisSetting(int type, int *value);
+
+/**
+ @brief Gets Simulation Analysis Setting
+ @param type Option code (see @ref SM_SimSetting)
+ @param[out] value Option value
+ @return Error code
+ */
+int DLLEXPORT swmm_getSimulationParam(int type, double *value);
+
+/**
+ @brief Gets Object Count
+ @param type Option code (see @ref SM_ObjectType)
+ @param[out] count Option value
+ @return Error code
+ */
+int DLLEXPORT swmm_countObjects(int type, int *count);
+
+/**
+ @brief Gets Object ID
+ @param type Option code (see @ref SM_ObjectType)
+ @param index of the Object
+ @param[out] id The string ID of object.
+ @return Error code
+ */
+int DLLEXPORT swmm_getObjectId(int type, int index, char **id);
+
+/**
+ @brief Gets Object Index
+ @param type Option code (see @ref SM_ObjectType)
+ @param[in] id of the Object
+ @param[out] index of the Object
+ @return errcode Error Code
+ */
+int DLLEXPORT swmm_getObjectIndex(SM_ObjectType type, char *id, int *index);
+
+/**
+ @brief Get the type of node with specified index.
+ @param index The index of a node
+ @param[out] Ntype The type code for the node (@ref SM_NodeType).
+ id must be pre-allocated by the caller.
+ @return Error code
+*/
+int DLLEXPORT swmm_getNodeType(int index, int *Ntype);
```

Comment:
Actually I would prefer to see something like 
```
int DLLEXPORT swmmgetNodeType(int index, SM_NodeType *type);
```

And then in swmm-python output.i use the following typemap.
```
/* TYPEMAPS FOR ENUMERATED TYPE OUTPUT ARGUMENTS */
%typemap(in, numinputs=0) EnumTypeOut * (int temp) {
    $1 = ($1_type)&temp;
}
%typemap(argout) EnumTypeOut * {
    char *temp = "$1_basetype";
    PyObject *module = PyImport_ImportModule("epanet.toolkit.solver_enum");
    PyObject *function = PyDict_GetItemString(PyModule_GetDict(module), (temp + 3));
    if (PyCallable_Check(function)) {
        PyObject *enum_out = PyObject_CallFunction(function, "i", *$1);
        %append_output(enum_out);
    }
}
```

Reply:
@michaeltryby, this will work for the enums. There are several instances where parameters to function start with an upper case. For instance, I see several instances of ```double **PollutArray```

Reply:
@cbuahin I can rename variables with camalcase. Is that ok?